web/cobrands/fixmystreet/assets.js

```diff
@@ -16,8 +16,58 @@ var fixmystreet = fixmystreet || {};
     };
 })();
 
+OpenLayers.Layer.VectorAsset = OpenLayers.Class(OpenLayers.Layer.Vector, {
+    initialize: function(name, options) {
+        OpenLayers.Layer.Vector.prototype.initialize.apply(this, arguments);
+        // Might only be able to fill in fields once they've been returned from the server
+        $(fixmystreet).on('report_new:category_change:extras_received', this.update_layer_visibility.bind(this));
+        // But also want to do it immediately in case it's hiding the form or something
+        $(fixmystreet).on('report_new:category_change', this.update_layer_visibility.bind(this));
+    },
+
+    update_layer_visibility: function() {
+        if (!fixmystreet.map) {
+          return;
+        }
+
+        if (!this.fixmystreet.always_visible) {
+            // Show/hide the asset layer when the category is chosen
+            var category = $('#problem_form select#form_category').val();
+            if (fixmystreet.assets.check_layer_relevant(this.fixmystreet, category)) {
+                this.setVisibility(true);
+                if (this.fixmystreet.fault_layer) {
+                    this.fixmystreet.fault_layer.setVisibility(true);
+                }
+                this.zoom_to_assets();
```

Comment:
As this is now firing on both category change and when extras are received, could this cause a confusion at some point? (But see other comment made later.)